kroxylicious-kubernetes-api/src/main/resources/META-INF/fabric8/kafkaproxies.kroxylicious.io-v1.yml

```diff
@@ -61,6 +61,90 @@ spec:
                             type: string
                           type: object
                       type: object
+                infrastructure:
+                  type: object
+                  description: |
+                    A collection of infrastructure-level settings that may be applied across all (or a subset of) 
+                    manifested resources for a KafkaProxy. This includes settings that affect resource behaviour at
+                    the platform layer, such as:
+                    - Resource limits and requests
+                    - Resource labels or annotations
+                    - Replica configuration
+                  properties:
+                    proxyDeployment:
+                      type: object
+                      description: |
+                        A specification for the desired behaviour of the proxy deployment.
+                      properties:
+                        replicas:
+                          type: integer
+                          description: "The number of pods to run."
+                          format: int32
+                          minimum: 0
+                          default: 1
+                        template:
+                          type: object
+                          properties:
+                            spec:
+                              type: object
+                              description: Specification of the desired behavior of the pod
+                              properties:
+                                containers:
```

Comment:
why do we have a containers node? we only really want to offer limits to the proxy pod so we can just assume a single node?

Reply:
I suppose we could use the pod resources concept: `allows you to declare an overall resource budget for the Pod, which is especially helpful when dealing with a large number of containers `

and our implementation is just a single container with the same resources set on it.

Reply:
just wondering what it would mean to have pod resources specified if we ever introduced another container, like a sidecar or whatever. the pod resources cover all containers.